libarchive/archive_read_support_format_tar.c

```diff
@@ -2156,10 +2158,21 @@ pax_attribute_LIBARCHIVE_xattr(struct archive_entry *entry,
 	/* Base-64 decode value */
 	value_decoded = base64_decode(value, value_length, &value_len);
 	if (value_decoded == NULL) {
+		archive_set_error(&a->archive, ARCHIVE_ERRNO_MISC,
+		    "Failed to decode xattr `%.*s'", (int)name_length, name);
 		free(name_decoded);
 		return 1;
 	}
 
+	if (value_len > xattr_limit) {
```

Comment:
I suppose my only concern is that this checks the length only _after_ the potential impact (denial of service, memory exhaustion, whatever the case may be) of having a too-large xattr. The allocation will complete or fail on line R2159 (`base64_decode`) and only then will we determine whether it was allowably-large.

For any value of length N encoded in base64, it will take up roughly `(N * 4) / 3` bytes.

Would it be possible to bail out before `base64_decode` by checking whether we have more base64-encoded data than would encode a value of length `xattr_limit`?